components/bl-category-chart/src/index.js

```diff
@@ -43,104 +17,166 @@ export default function CategoryChartComponent({ component, elRef }) {
       ref={ elRef }
       style={ styles }
       className={
-        cn("bl-customComponent-categoryChart", classList, { "bl-customComponent-categoryChart--disabled": disabled })
+        cn('bl-customComponent-categoryChart', classList, { 'bl-customComponent-categoryChart--disabled': disabled })
       }>
-      <canvas ref={ chartRef } style={{ backgroundColor: chartBackgroundColor, width: "100%", height: "100%" }} />
+      <canvas ref={ chartRef } style={{ backgroundColor, width: '100%', height: '100%' }}/>
     </div>
   );
 }
 
-const useChart = (chartRef, props) => {
+function useChartLogic(component) {
   const {
-    display, type, labels, datasets, options, chartTitleVisibility, chartTitleFontSize,
-    chartTitle, yGridLineVisibility, xGridLineVisibility, gridLinesColor, gridLinesWidth
-  } = props;
-  const chartInstance = useRef();
+    options, yGridLineVisibility, xGridLineVisibility, gridLinesColor, gridLinesWidth,
+    legendVisibility, titleVisibility, title, titleFontSize,
+    display, type, labels, datasets
+  } = component;
+
+  const chartRef = useRef(null);
+  const chartInstance = useRef(null);
+
+  const chartConfig = {
+    ...options,
+    responsive: true,
+    maintainAspectRatio: false,
+    plugins: {
+      legend: {
+        display: legendVisibility,
+      },
+      title: {
+        display: titleVisibility,
+        text: title,
+        font: {
+          size: titleFontSize,
+        },
+      },
+    },
+    scales: {
+      y: {
+        grid: {
+          display: yGridLineVisibility,
+          color: gridLinesColor,
+          lineWidth: gridLinesWidth,
+        },
+      },
+      x: {
+        grid: {
+          display: xGridLineVisibility,
+          color: gridLinesColor,
+          lineWidth: gridLinesWidth,
+        },
+      },
+    },
+  }
 
   useEffect(() => {
     if (!display) {
-      return null;
+      return;
+    }
+
+    let requireRerender = false;
+
+    const optionsConfig = { ...options };
+
+    if (isDataChange(chartConfig, optionsConfig)) {
+      requireRerender = true;
+      Object.assign(chartConfig, optionsConfig);
+    }
+
+    const scaleConfig = {
```

Comment:
scaleConfig and optionsConfig - it's not configs.
Use word `config` only for main config object